library/std/src/fs.rs

```diff
@@ -116,6 +116,19 @@ pub struct File {
     inner: fs_imp::File,
 }
 
+/// An enumeration of possible errors which can occur while trying to acquire a lock
+/// from the [`try_lock`] method and [`try_lock_shared`] method on a [`File`].
+///
+/// [`try_lock`]: File::try_lock
+/// [`try_lock_shared`]: File::try_lock_shared
+#[unstable(feature = "file_lock", issue = "130994")]
+pub enum TryLockError {
+    /// The lock could not be acquired due to an I/O error on the file.
+    Error(io::Error),
+    /// The lock could not be acquired at this time because the operation would block.
+    WouldBlock,
+}
```

Comment:
Should this document that `TryLockError::Error` containing `ErrorKind::WouldBlock` will never be returned by std?

Reply:
I think that makes sense, yes